swarm.go

```diff
@@ -395,15 +397,32 @@ func (s *Swarm) bestConnToPeer(p peer.ID) *Conn {
 		cLen := len(c.streams.m)
 		c.streams.Unlock()
 
-		if cLen >= bestLen {
-			best = c
-			bestLen = cLen
+		if isDirectConn(best) {
+			// Since the best connection we have till now is a direct connection,
+			// we will ONLY replace it with a direct connection
+			if !isDirectConn(c) {
+				continue
+			}
+			if cLen >= bestLen {
+				best = c
+				bestLen = cLen
+			}
+		} else {
+			// Since the best connection we have till now is a Relayed connection,
+			// we will simply replace with a newer connection if it has atleast as many streams.
+			if cLen >= bestLen {
+				best = c
+				bestLen = cLen
+			}
```

Comment:
Actually this is not correct: we always want to replace a relayed connection with a direct connection, regardless of how many streams it has.

Reply:
Ah, missed this. Have fixed it.